src/librbd/image/RemoveRequest.cc

```diff
@@ -69,6 +69,8 @@ Context *RemoveRequest<I>::handle_open_image(int *result) {
   if (*result < 0) {
     lderr(m_cct) << "error opening image: " << cpp_strerror(*result) << dendl;
     m_retval = *result;
+    if (*result != -ENOENT)
+      m_retval = *result;
```

Comment:
I still don't understand your fix. Have you tested this? This patch does not look like have any effect to me, because it still sets m_retval = *result one line above (and still proceed with the removal). And even looking at your #12638 I don't think this is what Jason suggests as a fix for this issue. I had impression he wanted we check for incompatible features, while you just make it to abort when it fails to open the image.

I wouldn't add the patch for this issue here until it is fixed separately.